packages/semcom-node/lib/store/services/pod-sync.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Handler, TypedKeyValueStore } from '@digita-ai/handlersjs-core';
+import { Parser } from 'n3';
+import { from, Observable, of } from 'rxjs';
+
+export class PodSyncService<S extends string, M extends { [s in S]: string[] }>
+  extends Handler<void, void> {
+
+  /**
+   *
+   * @param storage key in which the storage is located
+   * @param store the given store
+   */
+  constructor(
+    private readonly storage: S,
+    private readonly store: TypedKeyValueStore<M>,
+    private readonly localPod: string,
+  ) { super(); }
+
+  canHandle(): Observable<boolean> {
+
+    return of(true);
+
+  }
+
+  private async componentsInPod(uri: string): Promise<Set<string>> {
+
+    return new Promise((resolve) => {
+
+      // return an empty set if the pod is offline
+      fetch(uri, { headers: { 'Accept': 'text/turtle' } })
+        .then((response) => response.text()
+          .then((body) => {
+
+            resolve(new Set(
+              response.status === 200
+                ? new Parser({ format: 'Turtle' }).parse(body)
+                  .filter((quad) => quad.object.value === 'http://www.w3.org/ns/ldp#Resource')
+                  .filter((quad) => quad.subject.value !== '')
+                  .map((quad) => quad.subject.value)
+                : undefined
+            ));
+
+          }))
+        .catch(() => resolve(new Set()));
+
+    });
+
+  }
+
+  private async synchronizeComponents(): Promise<void> {
+
+    const existingComponents = await this.componentsInPod(this.localPod);
+
+    const pods: string[] = await this.store.get(this.storage) ?? [];
+
+    // map of components to be fatched, mapped to a node which has the component metadata
+    const componentToFetch: Map<string, string> = new Map();
+
+    await Promise.all(pods.map(async (pod) => {
+
+      [ ...await this.componentsInPod(pod) ]
+        .filter((component) => !existingComponents.has(component))
+        .forEach((component) => componentToFetch.set(component, pod));
```

Comment:
I think you can easily map this to a [component, pod] array, and then pass that to the Map constructor, instead of using `set`.

Reply:
I don't understand what you mean, the map should only be constructed once right so I can't call the constructor in a loop function, or do you mean something else?

Reply:
If you first create the array of pairs, you can then call Map(arrayOfPairs)

Reply:
i can't directly find a clean approach since they also need to be flattened but you can't use flat()